zarr/_storage/store.py

```diff
@@ -303,6 +348,125 @@ def _ensure_store(store):
         )
 
 
+class StorageTransformer(MutableMapping, abc.ABC):
+    """Base class for storage transformers. The methods simply pass on the data as-is
+    and should be overwritten by sub-classes."""
+
+    _store_version = 3
```

Comment:
Slight surprise to find this in store.py rather than under v3.

Reply:
I assumed that the base-classes go into `store.py`, similar to `StoreV3`, but also happy to move this into `_storage/v3.py` if that's a better fit.